features/lorawan/system/timer.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ / _____)             _              | |
+( (____  _____ ____ _| |_ _____  ____| |__
+ \____ \| ___ |    (_   _) ___ |/ ___)  _ \
+ _____) ) ____| | | || |_| ____( (___| | | |
+(______/|_____)_|_|_| \__)_____)\____)_| |_|
+    (C)2013 Semtech
+
+Description: Timer objects and scheduling management
+
+License: Revised BSD License, see LICENSE.TXT file include in the project
+
+Maintainer: Miguel Luis and Gregory Cristian
+
+
+Copyright (c) 2017, Arm Limited and affiliates.
+
+SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "timer.h"
+
+Timer TimeCounter;
```

Comment:
Why timer object is not C++ object ? Why are  Timer and Ticker in the global scope here? 

What is the need to have this Timer  ? I am missing the description for Timer

Reply:
Because LoRaMac.cpp doesn't follow any class structure and is not object oriented. This is one of the gruesome realities :)